webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -314,225 +318,9 @@ private static PdfReader createPdf(String globalTitle, Date creationDate, Scramb
-                cb.endText();
-
-
-                cb.beginText();
-                fontSize = 11;
-                cb.setFontAndSize(bf, fontSize);
-                cb.showTextAligned(PdfContentByte.ALIGN_CENTER, "DO NOT FILL IF YOU ARE THE COMPETITOR", competitorInfoLeft + (right-competitorInfoLeft)/2, top-offsetTop, 0);
-                offsetTop += fontSize + marginBottom;
-                cb.endText();
-
-                cb.beginText();
-                fontSize = 11;
-                cb.setFontAndSize(bf, fontSize);
-                cb.showTextAligned(PdfContentByte.ALIGN_CENTER, "Graded by: _______________ Result: ______", competitorInfoLeft + (right-competitorInfoLeft)/2, top-offsetTop, 0);
-                offsetTop += fontSize + marginBottom;
-                cb.endText();
-
-                cb.beginText();
-                cb.setFontAndSize(bf, 25f);
-                int MAGIC_NUMBER = 40; // kill me now
-                cb.showTextAligned(PdfContentByte.ALIGN_CENTER, "Fewest Moves", left+(competitorInfoLeft-left)/2, top-MAGIC_NUMBER, 0);
-                cb.endText();
-
-                com.itextpdf.text.List rules = new com.itextpdf.text.List(com.itextpdf.text.List.UNORDERED);
-                rules.add("Notate your solution by writing one move per bar.");
-                rules.add("To delete moves, clearly erase/blacken them.");
-                rules.add("Face moves F, B, R, L, U, and D are clockwise.");
-                rules.add("Rotations x, y, and z follow R, U, and F.");
-                rules.add("' inverts a move; 2 doubles a move. (e.g.: U', U2)");
-                rules.add("w makes a face move into two layers. (e.g.: Uw)");
-                rules.add("A [lowercase] move is a cube rotation. (e.g.: [u])");
-
-                ct.addElement(rules);
-                int rulesTop = competitorInfoBottom+55;
-                ct.setSimpleColumn(left+padding, scrambleBorderTop, competitorInfoLeft-padding, rulesTop, 0, Element.ALIGN_LEFT);
-                ct.go();
-
-                rules = new com.itextpdf.text.List(com.itextpdf.text.List.UNORDERED);
-                rules.add("You have 1 hour to find a solution.");
-                rules.add("Your solution length will be counted in OBTM.");
-                int maxMoves = WCA_MAX_MOVES_FMC;
-                rules.add("Your solution must be at most " + maxMoves + " moves, including rotations.");
-                rules.add("Your solution must not be directly derived from any part of the scrambling algorithm.");
-                ct.addElement(rules);
-                MAGIC_NUMBER = 150; // kill me now
-                ct.setSimpleColumn(left+padding, scrambleBorderTop, rulesRight-padding, rulesTop-MAGIC_NUMBER, 0, Element.ALIGN_LEFT);
-                ct.go();
-
-                doc.newPage();
+                addFMCSolutionSheet(docWriter, doc, scrambleRequest, globalTitle, i);
+                addFMCScrambleSheet(docWriter, doc, scrambleRequest, globalTitle, i);
```

Comment:
> It does not end up in the big pdf currently.

@gyroninja, what are you referring to when you say "it"? I just tested this code out, and I can confirm that the separate fmc scramble sheets *do* show up in the big PDF*. You are correct that the generic fmc solution sheet does not show up in the big PDF.

Call stack:

- [`requestsToPdf`](https://github.com/gyroninja/tnoodle/blob/c313716f3cb5eeef1daf55c68df8e55d59bfe556/webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java#L1229)
- [`createPdf`](https://github.com/gyroninja/tnoodle/blob/c313716f3cb5eeef1daf55c68df8e55d59bfe556/webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java#L221)
-  [`addScrambles`](https://github.com/gyroninja/tnoodle/blob/c313716f3cb5eeef1daf55c68df8e55d59bfe556/webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java#L323)